home-manager-module.nix

```diff
@@ -0,0 +1,100 @@
+{ systemdLib }:
+{
+  config,
+  osConfig,
+  options,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.virtualisation.user.quadlet;
+  quadletUtils = import ./utils.nix {
+    inherit lib;
+    systemdLib = systemdLib { inherit lib config pkgs; };
+  };
+  containerOpts = lib.types.submodule (import ./container.nix { inherit quadletUtils; });
+  networkOpts = lib.types.submodule (import ./network.nix { inherit quadletUtils pkgs; });
+in
+{
+  options.virtualisation.user.quadlet = {
```

Comment:
I followed the convention that exists in nixpkgs:

`systemd.service` -> `systemd.user.services`

But happy to change it.

Reply:
I think `systend.user.services` is based on that "user systemd" being something. We can have a service managed by "user systemd", or similarly a container managed by a concrete system named "user quadlet" / "user podman", rather than an abstract idea if "user virtualisation".

From the ownership perspective, the systemd module conceptually owns `systemd.**` options and does user vs system separation inside that scope, while quadlet-nix should probably do the same and implement the separation within its own land `virtualisation.quadlet.**`.

What do you think?

Reply:
Yes you've convinced me. I'll make the change!